samples/net/sockets/big_http_download/src/big_http_download.c

```diff
@@ -0,0 +1,255 @@
+{
+	while (len) {
+		ssize_t out_len = send(sock, buf, len, 0);
+		if (out_len < 0) {
+			return out_len;
+		}
+		buf = (const char *)buf + out_len;
+		len -= out_len;
+	}
+
+	return 0;
+}
+
+int skip_headers(int sock)
+{
+	int state = 0;
+
+	while (1) {
+		char c;
+		int st;
+
+		st = recv(sock, &c, 1, 0);
+		if (st <= 0) {
+			return st;
+		}
+
+		if (state == 0 && c == '\r') {
+			state++;
+		} else if (state == 1 && c == '\n') {
+			state++;
+		} else if (state == 2 && c == '\r') {
+			state++;
+		} else if (state == 3 && c == '\n') {
+			break;
+		} else {
+			state = 0;
+		}
+	}
+
+	return 1;
+}
+
+void print_hex(const unsigned char *p, int len)
+{
+	while (len--) {
+		printf("%02x", *p++);
+	}
+}
+
+void download(struct addrinfo *ai)
+{
+	int sock;
+
+	sock = socket(ai->ai_family, ai->ai_socktype, ai->ai_protocol);
+	CHECK(sock);
+	printf("sock = %d\n", sock);
+	CHECK(connect(sock, ai->ai_addr, ai->ai_addrlen));
+	sendall(sock, "GET /", SSTRLEN("GET /"));
+	sendall(sock, uri_path, strlen(uri_path));
+	sendall(sock, " HTTP/1.0\r\n", SSTRLEN(" HTTP/1.0\r\n"));
+	sendall(sock, "Host: ", SSTRLEN("Host: "));
+	sendall(sock, host, strlen(host));
+	sendall(sock, "\r\n\r\n", SSTRLEN("\r\n\r\n"));
+
+	if (skip_headers(sock) <= 0) {
+		printf("EOF or error in response headers\n");
+		goto error;
+	}
+
+	cur_bytes = 0;
+	mbedtls_md_starts(&hash_ctx);
+
+	while (1) {
+		int len = recv(sock, response, sizeof(response) - 1, 0);
+
+		if (len < 0) {
+			printf("Error reading response\n");
+			goto error;
+		}
+
+		if (len == 0) {
+			break;
+		}
+
+		mbedtls_md_update(&hash_ctx, response, len);
+
+		cur_bytes += len;
+		printf("%u bytes\r", cur_bytes);
+
+		response[len] = 0;
+		/*printf("%s\n", response);*/
+	}
+
+	printf("\n");
+
+	mbedtls_md_finish(&hash_ctx, response_hash);
+
+	printf("Hash: ");
+	print_hex(response_hash, mbedtls_md_get_size(hash_info));
+	printf("\n");
+
+	if (memcmp(response_hash, download_hash,
+		   mbedtls_md_get_size(hash_info)) != 0) {
+		printf("HASH MISMATCH!\n");
+	}
+
+error:
+	close(sock);
+}
+
+int main(void)
+{
+	static struct addrinfo hints;
+	struct addrinfo *res;
+	int st;
+	char *p;
+	unsigned int total_bytes = 0;
+
+	if (strncmp(download_url, "http://", SSTRLEN("http://")) != 0) {
+		fatal("Only http: URLs are supported");
+	}
+
+	p = download_url + SSTRLEN("http://");
+
+	/* Parse host part */
+	host = p;
+	while (*p && *p != ':' && *p != '/') {
+		p++;
+	}
+
+	/* Store optional port part */
+	if (*p == ':') {
+		*p++ = 0;
+		port = p;
+	}
+
+	/* Parse path part */
+	while (*p && *p != '/') {
+		p++;
+	}
+
+	if (*p == '/') {
+		*p++ = 0;
+		uri_path = p;
+	}
+
+	printf("Preparing HTTP GET request for http://%s:%s/%s\n",
+	       host, port, uri_path);
+
+	hints.ai_family = AF_INET;
+	hints.ai_socktype = SOCK_STREAM;
+	st = getaddrinfo(host, port, &hints, &res);
+	printf("getaddrinfo status: %d\n", st);
+
+	if (st != 0) {
+		fatal("Unable to resolve address");
+	}
+
+#if 0
```

Comment:
#if DEBUG, or some other DEBUG macro?

Reply:
Well, this code was just copied from the http_get sample, I guess, it can be just removed here.